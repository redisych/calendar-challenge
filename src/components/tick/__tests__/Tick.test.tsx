import * as React from 'react';
import * as enzyme from 'enzyme';
import { Tick } from '..';
import toJson from 'enzyme-to-json';

const getWrapper = (): any => {
    return enzyme.shallow(<Tick minute={60} />);
};

describe('Tick', () => {
    it('should be defined', () => {
        const wrapper = getWrapper();

        expect(wrapper).toBeDefined();
    });

    it('should match snapshot', () => {
        const wrapper = getWrapper();

        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('should render time mark if is an hour', () => {
        const wrapper = getWrapper();

        expect(wrapper.find('.time-mark').exists()).toBeTruthy();
    });

    it('should not render time mark if is half an hour', () => {
        const wrapper = getWrapper();

        wrapper.setProps({
            minute: 30
        });

        expect(wrapper.find('.time-mark').exists()).toBeFalsy();
    });

    it('should not render time label with className "time-label-half-hour" if is an hour', () => {
        const wrapper = getWrapper();

        expect(wrapper.find('.time-label').hasClass('time-label-half-hour')).toBeFalsy();
    });

    it('should render time label with className "time-label-half-hour" if is a half an hour', () => {
        const wrapper = getWrapper();

        wrapper.setProps({
            minute: 30
        });

        expect(wrapper.find('.time-label').hasClass('time-label-half-hour')).toBeTruthy();
    });
});
