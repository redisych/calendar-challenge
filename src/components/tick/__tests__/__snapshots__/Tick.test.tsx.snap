// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tick should match snapshot 1`] = `
<div
  className="tick"
>
  <span
    className="time-label"
  >
    10:00
  </span>
  <span
    className="time-mark"
  >
    AM
  </span>
</div>
`;
